OIIO/WriteOIIO.cpp

```diff
@@ -1099,10 +1106,18 @@ WriteOIIOPlugin::beginEncodeParts(void* user_data,
     }
 #endif // ifdef OFX_IO_USING_OCIO
     if (!_quality->getIsSecret()) {
+#if OIIO_VERSION >= 20316
+        compression += ':' + std::to_string(quality);
+#else
         spec.attribute("CompressionQuality", quality);
+#endif
     }
     if (!_dwaCompressionLevel->getIsSecret()) {
+#if OIIO_VERSION >= 20316
+        compression += ':' + std::to_string((int)dwaCompressionLevel);
```

Comment:
After looking a little closer at the OpenImageIO code (https://github.com/OpenImageIO/oiio/blob/master/src/libOpenImageIO/formatspec.cpp#L1187), I  believe you could just use spec.attribute("CompressionQuality", (int)dwaCompressionLevel) here and keep the old code that used it above. Then we don't have to and encode/decode these values from strings.

Reply:
yes, but support for CompressionQuality may disappear at any time, since it's here for backward compatibility and it's not documented anymore, see https://github.com/OpenImageIO/oiio/pull/2111

Reply:
Fair enough. Thanks for the link.